dataland-backend/src/main/kotlin/org/dataland/datalandbackend/controller/DataController.kt

```diff
@@ -35,16 +36,22 @@ abstract class DataController<T>(
     private val logger = LoggerFactory.getLogger(javaClass)
     private val logMessageBuilder = LogMessageBuilder()
 
-    override fun postCompanyAssociatedData(companyAssociatedData: CompanyAssociatedData<T>):
+    override fun postCompanyAssociatedData(companyAssociatedData: CompanyAssociatedData<T>, bypassQa: Boolean):
         ResponseEntity<DataMetaInformation> {
+        if (bypassQa && !canUserBypassQa(DatalandAuthentication.fromContextOrNull())) {
+            throw AccessDeniedException(logMessageBuilder.bypassQaDeniedExceptionMessage)
+        }
         val companyId = companyAssociatedData.companyId
         val reportingPeriod = companyAssociatedData.reportingPeriod
         val userId = DatalandAuthentication.fromContext().userId
         val uploadTime = Instant.now().toEpochMilli()
         logger.info(logMessageBuilder.postCompanyAssociatedDataMessage(userId, dataType, companyId, reportingPeriod))
         val correlationId = generateCorrelationId(companyAssociatedData.companyId)
         val datasetToStore = buildDatasetToStore(companyAssociatedData, userId, uploadTime)
-        val dataIdOfPostedData = dataManager.addDataSetToTemporaryStorageAndSendMessage(datasetToStore, correlationId)
+        val dataIdOfPostedData = dataManager.addDataSetToTemporaryStorageAndSendMessage(
+            datasetToStore,
+            bypassQa, correlationId,
+        )
         logger.info(logMessageBuilder.postCompanyAssociatedDataSuccessMessage(companyId, correlationId))
         return ResponseEntity.ok(
```

Comment:
Bei `getCompanyAssociatedData()` unten schmeißen wir folgende message zurück, wenn der isViewable-Check `false` zurückwirft: 
`"You are trying to access a unreviewed dataset."`

Das stimmt in manchen Fällen, aber in manchen auch nicht.  Es kann ja jetzt vielfältige Gründe haben, wieso das dataset nicht anschaubar ist (sieht man ja wenn man sich die `isDatasetViewableByUser()` Methode anschaut).

Ich finde daher, dass die AccessDenied message entweder generisch sein muss, oder Infos über den Fall zurückgeben sollte, in dem man gerade ist, z.B. :
- Sorry aber das dataset ist noch pending review
- Sorry aber das deteset wurde rejected und ist daher nur vom Uploader oder von Reviewern viewable.
-

Reply:
Ist nun spezialisierter.

Reply:
Finde ich gut.